Sources/App/Controllers/API/Package+Badge.swift

```diff
@@ -3,10 +3,26 @@ import Vapor
 
 extension Package {
 
+    enum CompatibilityResult<Value> {
+        case available([Value])
+        case pending
+        
+        var values: [Value] {
+            if case .available(let values) = self {
+                return values
+            }
+            
+            return []
+        }
+    }
+    
     /// Returns a list of compatible swift versions across a package's significant versions.
     /// - Returns: Array of compatible `SwiftVersion`s
-    func swiftVersionCompatibility() -> [SwiftVersion] {
-        let builds = allSignificantBuilds()
+    func swiftVersionCompatibility() -> CompatibilityResult<SwiftVersion> {
+        let allBuilds = allSignificantBuilds()
+        if allBuilds.anyPending { return .pending }
```

Comment:
So one questions I have is this: do we want to make the whole value pending if just a single item is pending? If for some reason a job keeps dying the whole item will be pending.

Compatibility is a "wide" status item in that it aggregates across platforms and Swift versions: if any platform support a given swift version it's called compatible with that Swift version, and the same for platforms. The way I think of it is a logical "OR" across those two dimensions.

It feels like "pending" should also be "OR" and only "win" if all items are "pending". Chances are quite high that a single stuck build will be covered via the "OR"s across the Swift versions and platforms and "pending" would be unnecessarily strict.

Does that make sense?

Reply:
So to clarify, you'd rather change this from `if any pending` to `if all pending`?

Reply:
I _think_ so but I haven't actually played with this in earnest. I was looking at it through the lens of the issue report: we had a package with all builds pending and showed "unavailable". I think we want to show "pending" in that case and this patch would do that.

However, I remember that as soon as a few builds came through, the matrix started filling out (it does to quickly, because of the "OR", even when a number of builds are still missing) and I feel like we should reflect that in the badge as well.

Just wanted to open the discussion on that while it's warm!

Reply:
From a technical perspective, either are possible and super quick so happy to take a steer from you folks on that!

My personal opinion is that an incomplete badge could be misleading/confusing to users within the context of a README, and I'd rather see a badge which says "pending" rather than suggest I don't support a platform the user is looking for.

As somebody purely looking at a README, if these badges are a first impression and answer a quick question of compatibility without expecting the user to read much into it... I could've just lost a user!

Personal opinion, as always, your platform 😄

Reply:
That's a good point. The problem I see though is that a correct badge will be pending until all builds are complete even when it's already "filled out".

Say for instance all builds except the Linux 5.0 build are done: the whole build is pending and yet the badge would already correctly show Linux support, because the 5.2+ builds have completed. I.e. we'd "grey out" the badge until the slowest job is complete even though they late builds very likely won't add any more info.

Perhaps we could make this

```
[iOS | macOS | (some pending)]
```

but only if we lack info to fill it out. I.e. if we have a lagging build but already know from the other Swift versions that a platform is supported (likewise for Swift versions via other platforms) then we don't mark it pending.

Reply:
I'm thinking of this in terms of projecting the compatibility matrix down to the two axes for the badge labels:
 
<img width="822" alt="CleanShot 2021-07-05 at 17 22 42@2x" src="https://user-images.githubusercontent.com/65520/124493187-a0b57e00-ddb5-11eb-8e21-0203afcf1359.png">

Only if there's no "ok" in a column or row do we mark it as pending.

Reply:
And to contrast this to the PR as I understand it, that would display this instead:

<img width="835" alt="CleanShot 2021-07-05 at 17 26 22@2x" src="https://user-images.githubusercontent.com/65520/124493659-2f29ff80-ddb6-11eb-983b-75723994d2b3.png">

Reply:
> Only if there's no "ok" in a column or row do we mark it as pending.

I was just typing this up! This is exactly how it should work 👍

Reply:
Addendum: or if they're all failed - in case it's not pending either of course!

Reply:
It's worth saying if they all come back as failed, with both the current and this PR's version we would return "unavailable". Is this still fair? Or do we also now want an error case too?

Reply:
I think that’s fair. The badge only shows “supported” and an “all failed” is absence of the particular col/row. If there’s no “supported” at all for any row/col, “unavailable” seems like the only thing we can sensibly display (as I don’t think there’s a way to withdraw the whole badge).

Reply:
Switched to check for all satisfy pending, if we believe it'd make more sense to make it (as you suggested above) like `[iOS | macOS | (some pending)]` then let me know and I can follow that up.

Reply:
Looks good to me, thanks a lot, James!

I don't think we need to bother with the `(some pending)` for the moment. If I'm not mistaken we've simply omitted incomplete builds all along so this is not a regression.

We can always follow up if we change our minds.